src/file-handlers.ts

```diff
@@ -268,13 +283,28 @@ function renderFile(
  * ```
  */
 
+function renderFileAsync(
+  filename: string,
+  data: DataObj,
+  config?: PartialConfig
+): Promise<string>
+
+function renderFileAsync(
+  filename: string,
+  data: DataObj,
+  config: PartialConfig,
```

Comment:
I wonder if we should change this to `config?:`, and then delete lines 299-304

Reply:
We can't mark the parameter as optional using `?` here, since it is followed by another, in that case, required parameter, which is `cb`. What would be possible is to allow undefined (`PartialConfig | undefined`). The types would still be correctly inferred in each [scenario mentioned above](https://github.com/eta-dev/eta/pull/199#issuecomment-1353727149).

However, I am not too sure about removing line 299-304, mainly because I think this introduces a breaking change for certain use cases. To give a theoretical example:

```ts
const optionalCallbackFn: CallbackFn | undefined = ...;
Eta.renderFileAsync(fileName, { ...data, ...config }, callbackFn);
```

Would not compile if we decide to remove 299-304. If we keep it, it would compile and the type would be inferred as `Promise<string> | void`.

What do you think?